daemon/src/projection.rs

```diff
@@ -751,6 +751,18 @@ impl Cfd {
 
         in_open_state && not_expired
     }
+
+    fn is_closed(&self) -> bool {
```

Comment:
❓

So basically you are mapping the states to an "aggregated" state of `is_open`, `is_closed` and `is_failed`. Why are we doing this based on the projection states and not the events? 
We *can* do this, but I am not sure it's a good idea. I feel we add one more layer of mapping.

That goes back to the discussion if we should keep the `projection::CfdState` or change / remove it. How do you see this evolve?

---

Are you planning to expose the new "states" (open, closed, failed) on the API?
So far they are only used in tests.

Reply:
Yes, the idea is to have a simple function which allows us to determine if a position is `open`, `closed` or `failed`. 

> Why are we doing this based on the projection states and not the events?

Do you think based on the events is better?

> Are you planning to expose the new "states" (open, closed, failed) on the API?

No, it should only be used for the metrics.

Reply:
> [snip]
> No, it should only be used for the metrics.

How are you planning to expose the metrics? As another route that then pulls them from the projection?

Reply:
The maker daemon has already a metric endpoint under `api/metrics`: 
The _source_ of this endpoint is managed by prometheus and used here: 
https://github.com/itchysats/itchysats/blob/ec30d46f04caaa7cec3dc366e21598545c75dd53/maker/src/routes.rs#L275-L286

Reply:
Maybe worth to mention: the metrics are static variables stored in a global registry. `prometheus.gather()` knows how to find them.

Reply:
Right, I think I never looked into this code in detail. I will have another broader look later to come up with a more complete conclusion.